File: eBank.DataAccess/Models/Customer/CustomerModel.cs
using eBank.DataAccess.Models.Base;
using eBank.DataAccess.Models.User;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eBank.DataAccess.Models
{
    public class CustomerModel : BaseModel
    {
        [Key]
        public int CustomerId { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public DateTime DateOfBirth { get; set; }

        public int LegalStatus { get; set; }

        public int MaritalStatus { get; set; }

        [ForeignKey("ApplicationUser")]
        public string UserId { get; set; }
        public virtual ApplicationUser ApplicationUser { get; set; }
    }
}

File: eBank.Web/ClientApp/src/app/customer/models/customer.model.ts
import { ICustomer } from '../interfaces/customer.interface';

export class Customer implements ICustomer {
    id: number;
    customerId: number;
    firstName: string;
    lastName: string;
    dateOfBirth: Date;
    legalStatus: number;
    maritalStatus: number;
    userId: string;
}
